File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int MaxDepth(TreeNode root) {
        if (root == null) return 0;
        if (root.left == null && root.right == null) return 1;
        int left = root.left != null ? MaxDepth(root.left) : 0;
        int right = root.right != null ? MaxDepth(root.right) : 0;
        return Math.Max(left, right) + 1;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func kthSmallest(root *TreeNode, k int) int {
    var visited []*TreeNode
	
	var inorder func(*TreeNode)
    inorder = func(node *TreeNode) {
		if node == nil {
			return
		}
		inorder(node.Left)
		visited = append(visited, node)
		inorder(node.Right)
	}
	
	inorder(root)
	return visited[k-1].Val
}
